### PR TITLE
MINOR: cleanup Kafka Streams exception classes

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

**Comment by mjsax** [comment]
@guozhangwang @dguy @enothereska @miguno 

This is part of the global JavaDoc and code cleanup we want to do. I will open many small PRs to keep reviewing overhead small.

**Comment by mjsax** [comment]
Btw: I think we might want to add more exception to improve overall error handling in the whole code base.

File: streams/src/main/java/org/apache/kafka/streams/errors/InvalidStateStoreException.java [hunk 1]
```diff
@@ -14,18 +14,31 @@
...
+@InterfaceStability.Unstable
 public class InvalidStateStoreException extends StreamsException {
 
+    private final static long serialVersionUID = 1L;
+
     public InvalidStateStoreException(final String message) {
         super(message);
     }
+
+    public InvalidStateStoreException(final String message, final Throwable throwable) {
```
Comment (dguy) [inline streams/src/main/java/org/apache/kafka/streams/errors/InvalidStateStoreException.java:37]: Why add this and the next ctor? They aren't used anywhere. We should add them when/if they are needed

File: streams/src/main/java/org/apache/kafka/streams/errors/LockException.java [hunk 2]
```diff
@@ -14,22 +14,27 @@
...
 
-    public LockException(final String s) {
-        super(s);
+    public LockException(final String message) {
+        super(message);
     }
 
-    public LockException(String s, Throwable throwable) {
-        super(s, throwable);
+    public LockException(final String message, final Throwable throwable) {
```
Comment (dguy) [inline streams/src/main/java/org/apache/kafka/streams/errors/LockException.java:33]: Same as previous comment.

File: streams/src/main/java/org/apache/kafka/streams/errors/ProcessorStateException.java [hunk 3]
```diff
@@ -17,22 +17,27 @@
...
     }
 
-    public ProcessorStateException(String s, Throwable throwable) {
-        super(s, throwable);
+    public ProcessorStateException(final String message, final Throwable throwable) {
+        super(message, throwable);
     }
 
-    public ProcessorStateException(Throwable throwable) {
+    public ProcessorStateException(final Throwable throwable) {
```
Comment (dguy) [inline streams/src/main/java/org/apache/kafka/streams/errors/ProcessorStateException.java:40]: As above

File: streams/src/main/java/org/apache/kafka/streams/errors/TaskAssignmentException.java [hunk 4]
```diff
@@ -17,21 +17,23 @@
...
     }
 
-    public TaskAssignmentException(String s, Throwable throwable) {
-        super(s, throwable);
+    public TaskAssignmentException(final String message, final Throwable throwable) {
+        super(message, throwable);
     }
 
-    public TaskAssignmentException(Throwable throwable) {
+    public TaskAssignmentException(final Throwable throwable) {
```
Comment (dguy) [inline streams/src/main/java/org/apache/kafka/streams/errors/TaskAssignmentException.java:36]: As above

File: streams/src/main/java/org/apache/kafka/streams/errors/TaskIdFormatException.java [hunk 5]
```diff
@@ -17,14 +17,28 @@
...
 
     private static final long serialVersionUID = 1L;
 
-    public TaskIdFormatException(String taskString) {
-        super("Task id cannot be parsed correctly" + (taskString == null ? "" : " from " + taskString));
+    public TaskIdFormatException(final String message) {
+        super("Task id cannot be parsed correctly" + (message == null ? "" : " from " + message));
+    }
+
+    public TaskIdFormatException(final String message, final Throwable throwable) {
```
Comment (dguy) [inline streams/src/main/java/org/apache/kafka/streams/errors/TaskIdFormatException.java:37]: As above for this and the next ctor

File: streams/src/main/java/org/apache/kafka/streams/errors/TopologyBuilderException.java [hunk 6]
```diff
@@ -17,14 +17,26 @@
...
 public class TopologyBuilderException extends StreamsException {
 
     private static final long serialVersionUID = 1L;
 
-    public TopologyBuilderException(String message) {
+    public TopologyBuilderException(final String message) {
         super("Invalid topology building" + (message == null ? "" : ": " + message));
     }
+
+    public TopologyBuilderException(final String message, final Throwable throwable) {
```
Comment (dguy) [inline streams/src/main/java/org/apache/kafka/streams/errors/TopologyBuilderException.java:35]: As above for this and next ctor

**Review by dguy — COMMENTED** [review]
Left some comments w.r.t unused constructors. I don't see the point of adding unused code.

**Comment by mjsax** [comment]
@dguy When I was adding `LockException` in another PR, I was explicitly requested to add more ctors even if I did not use them (see https://github.com/apache/kafka/pull/2233/files#r92231489)

**Comment by guozhangwang** [comment]
> was explicitly requested to add more ctors even if I did not use them

I am actually preferring to have the ctors in place even if they are not used yet. But again this is pure-subjective.

**Comment by dguy** [comment]
I'm not overly concerned, however generally we shouldn't write code that is not used. So my preference is to remove them unless they are needed. Dead code is dead code. But, whatever...

**Comment by mjsax** [comment]
I share @dguy opinion of avoiding dead code -- but I don't care too much either. @guozhangwang I guess it's your call to make a decision.

**Comment by guozhangwang** [comment]
To me it is case-by-case: I would not worry too much about such dead code in exception classes; but if it is some functional logic in, say `StreamThread` I would definitely remove it.

**Comment by guozhangwang** [comment]
Merged to trunk, thanks.

**Pull request closed without merging**